python/ray/data/dataset_iterator.py

```diff
@@ -103,6 +104,31 @@ def iter_batches(
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]:
+        """Return a local row iterator over the dataset.
+
+        If the dataset is a tabular dataset (Arrow/Pandas blocks), dict-like mappings
+        :py:class:`~ray.data.row.TableRow` are yielded for each row by the iterator.
+        If the dataset is not tabular, the raw row is yielded.
+
+        Examples:
+            >>> import ray
+            >>> for i in ray.data.range(1000000).iterator().iter_rows(): # doctest: +SKIP
+            ...     print(i) # doctest: +SKIP
```

Comment:
I know this example is copied from elsewhere, but it isn't tested. Also, if you remove `# doctest: +SKIP` this makes Sphinx break.
```suggestion
            >>> import ray
            >>> dataset = ray.data.range(10)
            >>> next(iter(dataset.iterator().iter_rows()))
            1
```

Reply:
Updated